src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -198,5 +211,18 @@ private string GetDebuggerDisplay()
         {
             return nameof(Path) + " = " + Path;
         }
+
+        private static void ValidateFileLength(Workspace workspace, string path)
+        {
+            var fileLength = FileUtilities.GetFileLength(path);
+            if (fileLength > workspace.Options.GetOption(FileTextLoaderOptions.FileLengthThreshold))
+            {
+                // log max file length to AI
+                Logger.Log(FunctionId.FileTextLoader_FileLengthThresholdExceeded, KeyValueLogMessage.Create(m => m["FileLength"] = fileLength));
+
+                var message = string.Format(WorkspacesResources.File_size_exceeds_maximum_allowed_size_0, path);
+                throw new IOException(message);
```

Comment:
Why IOExecption?  Why not just throw Execption.  Or InvalidOperationException.  No IO is actually occuring here.